python/lsst/meas/algorithms/defects.py

```diff
@@ -151,6 +151,33 @@ def __eq__(self, other):
     def __str__(self):
         return "Defects(" + ",".join(str(d.getBBox()) for d in self) + ")"
 
+    def normalize(self):
+        """Recalculate defects bounding boxes to use minimal set.
```

Comment:
I'm not sure this is true.

Specifically, I don't think that the normalized bounding boxes are necessarily a minimal set.

You can check this yourself, based on `DefectsTestCase.testLsstTextFile`:

```python
        with lsst.utils.tests.getTempFilePath(".txt") as tmpFile:
            with open(tmpFile, "w") as fh:
                print("""# X0  Y0  width height
     996        0       56       24
       0     4156     2048       20
       0        0       17     4176
    1998     4035       50      141
    1023        0        2     4176
    2027        0       21     4176
       0     4047       37      129
# Some rows without fixed column widths
14 20 2000 50
10 10 10 10
""", file=fh)

            defects = algorithms.Defects.readLsstDefectsFile(tmpFile)
```

At this point:

```python
>>> len(defects)
9
>>> len(defects.normalize())
11
```

Based on a quick visual inspection, I think the normalized set is equivalent to the inputs, but it's obviously not “minimal”.

Thoughts on whether this is actually a problem? I suspect the answer is “no”; we shouldn't assert that it is minimal, but it should be less pathological in most cases, and that's probably all we need. Right? Thoughts? @timj  @plazas ?

Reply:
That's very interesting. It implies that `Defects.fromMask` is not as clever as we thought it would be. It would be a fun exercise to plot the different bounding boxes to see how they differ but that should be a whole new ticket that looks into why `fromMask` is not optimal.  We have no other option in the short term though, so tweaking the documentation is the best option -- it does imply that if we fix fromMask we would get a better but different normalization.

Reply:
DM-24781.

Reply:
@plazas please drop a `TODO: DM-24781` in the updated docs to note this pending ticket.